packages/cli-platform-ios/src/commands/runIOS/index.ts

```diff
@@ -573,6 +579,50 @@ function getProcessOptions({
   };
 }
 
+function getBuildConfigurationFromXcScheme(
+  {scheme, configuration}: FlagsT,
+  sourceDir: string,
+) {
+  try {
+    const xcProject = fs
+      .readdirSync(sourceDir)
+      .find((dir) => dir.includes('.xcodeproj'));
+
+    if (xcProject) {
+      const xmlScheme = fs.readFileSync(
+        path.join(
+          sourceDir,
+          xcProject,
+          'xcshareddata',
+          'xcschemes',
+          `${scheme}.xcscheme`,
+        ),
+        {
+          encoding: 'utf-8',
+        },
+      );
+
+      const {Scheme} = xmlParser.parse(xmlScheme);
+
+      return Scheme.LaunchAction['@_buildConfiguration'];
+    }
+  } catch {
+    logger.error(`Could not find scheme ${scheme}.`);
```

Comment:
I think this should throw. 
When you pass dummy scheme it will log error but still proceed with the build:
```bash 
❯ node ../cli/packages/cli/build/bin.js run-ios --scheme Foo                      
error Could not find scheme Foo.
info Found Xcode workspace "Bar.xcworkspace"
info Building (using "xcodebuild -workspace Bar.xcworkspace -configuration  -scheme Foo -destination id=D83F179F-6C0B-45BA-9104-45397BA3FFB9")
```

Reply:
> I think this should throw.
> When you pass dummy scheme it will log error but still proceed with the build:

Thanks for catching that, done!

> And there's also a case when user passes good scheme (or no scheme at all and we select default one) but not existing configuration... Do you think we can detect this as well?
> (Although this is neither scope of this PR and is also working like that with current state of CLI, so we can fix it as follow up or move responsibility to pass good configuration to the user 😉 )

If i understand correctly, the XCode allows only two configurations: Debug and Release. I think the easiest way to detect it is to just check if configuration value is one of those and throw error/run in Debug as default if it's not correct. But agree - it's not in the scope of this PR so we can fix it separately if we decide what solution fits us best

Reply:
> If i understand correctly, the XCode allows only two configurations: Debug and Release

Not really, you can have custom ones I think as well :)

Reply:
Yes, you can have others